shared/health-bindings/src/main/java/io/pravega/shared/health/bindings/resources/HealthImpl.java

```diff
@@ -53,12 +49,10 @@
 
     private final HealthEndpoint endpoint;
 
-    private final RESTAuthHelper restAuthHelper;
     private final AuthorizationResource authorizationResource = new AuthorizationResourceImpl();
 
-    public HealthImpl(AuthHandlerManager pravegaAuthManager, HealthEndpoint endpoint) {
```

Comment:
Do we really want to get rid of _all_ authentication, or just authentication for the readiness/liveness routes so the operator(s) can use them?

Reply:
I think that was the idea yes. BUT, to add more nuance, that idea came with the caveat of "are we sure _all_ of these are ok to expose unauthenticated"?  (maybe the answer yes, it just needs to be explicitly decided)

I see in the API spec some liveness/health, then some more detailed calls.    Depending what is returned from these more detailed ones, you may or may not want to leave it _all_ open, I'm not sure.

For sure the get /liveness , /readiness, /status, look very safe.    I don't know enough about the rest of them though.

It IS possible to secure some and not others (by passing the authManager still, but removing its use only where you want).

@SrishT , just for playing devil's advocate, what does some of these "detail" calls return in terms of payload?

Reply:
We don't have any of them yet. We will be building them gradually both in the Controller and Segment Store. 

In the Segment Store, for example, details may include:
- Each assigned segment container and its status (running, starting, etc.)
- For each container:
    - Each segment active on a segment container (this might be a lot - but we'll see what we can do). We will include segment name, length sealed/etc.
    - Status of each internal Segment Container component (storage writer, durable log, metadata cleaner, event processor, etc.) 
- Quick stats about the cache.

In essence, the only customer information that will be exposed is the segment names. From those you can derive stream and scope names (but remember that segments are internally generated). The only information that this could provide to an "attacker" would be that "segment X exists and is part of stream Y". But then such as hypothetical attacker would still need to go through our regular gateway to access more information about such segment or read its contents, etc. 

The goal of this is not to expose too many details or pry-open the segments themselves (we have the Admin CLI under works for that goal), but to get a quick glance as to what's running within normal parameters, what's running hot and what's not running. To give you an idea of how this might be used, we may decide to do a sys admin dashboard that periodically pings all our services with "details == true" and present a one-page visual about the health of the cluster; that page would allow the admin to immediately see if something is requiring immediate attention or not. 

This should not really contain any more information than we already publish in our metrics - we already publish stream and segment tags there.

Reply:
@sarlaccpit currently the `/details` endpoint does not return anything.
But if there is any plan of exposing any sensitive information from this endpoint which unauthenticated users should not be given access to, then in that case it will be worthwhile to expose this particular endpoint only to authenticated users.
But since I am not sure about how sensitive these details would be, I would like to hear from @andreipaduroiu  and @pbelgundi about your opinion on this.

Reply:
@SrishT I think @andreipaduroiu has tried to answer your question about what the "details" API could contain. Itoo  don't have complete clarity on what this API would return for Controller, but to play the devils advocate I would not recommend returning this to an unauthenticated user. As suggested by @sarlaccpit "/liveness, /readiness and & status" should be fine being unauthenticated.